journal/journal.go

```diff
@@ -53,11 +53,18 @@ const (
 var conn net.Conn
 
 func init() {
+	Reconnect()
+}
+
+// Estabish a new connection to the journald socket and returns a boolean
+// indicating if it has succeeded or failed.
+func Reconnect() bool {
 	var err error
 	conn, err = net.Dial("unixgram", "/run/systemd/journal/socket")
```

Comment:
I think there may be a leak here now, as nobody calls Close on the previous conn.